spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarReactiveProperties.java

```diff
@@ -835,11 +834,8 @@ public ReactiveMessageConsumerSpec buildReactiveMessageConsumerSpec() {
 			map.from(this::getTopicsPattern).to(spec::setTopicsPattern);
 			map.from(this::getSubscriptionName).to(spec::setSubscriptionName);
 			map.from(this::getSubscriptionType).to(spec::setSubscriptionType);
-			map.from(this::getKeySharedMode).as((mode) -> switch (mode) {
-				case STICKY -> KeySharedPolicy.stickyHashRange();
-				case AUTO_SPLIT -> KeySharedPolicy.autoSplitHashRange();
-			}).to(spec::setKeySharedPolicy);
```

Comment:
Yeh, I can envision something like the following. It would require a custom `Range` converter to take `1,100` -> Range(1, 100).

```yaml
spring.pulsar.reactive.consumer.key-shared:
  mode: sticky | auto
  ranges: 
    - 1,100
    - 101,199
    - 1000-1500
```

Reply:
👍